econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -0,0 +1,1044 @@
+                inf = est.const_marginal_ate_inference(X=X_xf)
+                global_inference = NormalInferenceResults(d_t=inf.d_t, d_y=inf.d_y,
+                                                          pred=inf.mean_point,
+                                                          pred_stderr=inf.stderr_mean,
+                                                          mean_pred_stderr=None,
+                                                          inf_type='ate')
+
+            # Set the dictionary values shared between local and global summaries
+            if discrete_treatment:
+                cats = est.transformer.categories_[0]
+                baseline = cats[est.transformer.drop_idx_[0]]
+                cats = cats[np.setdiff1d(np.arange(len(cats)),
+                                         est.transformer.drop_idx_[0])]
+                d_t = len(cats)
+                insights = {
+                    _CausalInsightsConstants.TypeKey: ['cat'] * d_t,
+                    _CausalInsightsConstants.RawFeatureNameKey: [name] * d_t,
+                    _CausalInsightsConstants.CategoricalColumnKey: cats.tolist(),
+                    _CausalInsightsConstants.EngineeredNameKey: [
+                        f"{name} (base={baseline}): {c}" for c in cats]
+                }
+            else:
+                d_t = 1
+                cats = ["num"]
+                baseline = None
+                insights = {
+                    _CausalInsightsConstants.TypeKey: ["num"],
+                    _CausalInsightsConstants.RawFeatureNameKey: [name],
+                    _CausalInsightsConstants.CategoricalColumnKey: [name],
+                    _CausalInsightsConstants.EngineeredNameKey: [name]
+                }
+            result = CausalAnalysis._results(feature_index=feat_ind,
+                                             feature_name=name,
+                                             feature_baseline=baseline,
+                                             feature_levels=cats,
+                                             hinds=hinds,
+                                             X_transformer=X_transformer,
+                                             W_transformer=W_transformer,
+                                             estimator=est,
+                                             global_inference=global_inference)
+
+            return insights, result
+
+        if self.feature_names is None:
+            if hasattr(X, "iloc"):
+                feature_names = X.columns
+            else:
+                feature_names = [f"x{i}" for i in range(X.shape[1])]
+        else:
+            feature_names = self.feature_names
+
+        self.feature_names_ = feature_names
+
+        # extract subset of names matching new columns
+        new_feat_names = _safe_indexing(feature_names, new_inds)
+
+        cache_updates = dict(zip(new_inds,
+                                 joblib.Parallel(n_jobs=self.n_jobs,
```

Comment:
Most prob we should be using a minimum of n_jobs and the number of new_inds.

Reply:
hm I run it with 1 job and it still was saying: starting joblib with 8 concurrent workers.

Reply:
Yeah, you're right; on the other hand I'm not sure how much of a problem it is to have short lived workers that don't receive any work.  Going to punt for now given more pressing issues.

Reply:
Yea let's put that in the list of todos then